### DISCUSSION TITLE
colorPrimary如何覆盖modifyVars: v4Token

### DISCUSSION THREAD

**Discussion by xiaomoziyi** [ant-design/ant-design#45315] [Q&A]
![](https://gitee.com/xiaomoziyi/blogImage/raw/master/img/20231013105816.png)

![](https://gitee.com/xiaomoziyi/blogImage/raw/master/img/20231013105853.png)

![](https://gitee.com/xiaomoziyi/blogImage/raw/master/img/20231013105923.png)

lessloader的modifyVars的优先级为什么比在theme或者configProvider里的高，针对按钮配置色好像是生效的，但是a标签的颜色是不生效的，还是默认的`#1677ff`
![](https://gitee.com/xiaomoziyi/blogImage/raw/master/img/20231013110226.png)

**Comment by MadCcc** [comment]
modifyVars 和 theme 是相互隔离的，互不影响

**Reply by xiaomoziyi** [reply]
按照v4,只要设置了primaryColor是全局生效的，但是现在按照官方给的方案，antd升级到5，针对less中引用了antd变量问题，通过使用modifyVars来解决，可是这样，primaryColor就无法自定义了。有没有方案在defaultAlgorithm或者convertLegacyToken就解决呢
